<div style="display: flex; align-items: center;">
    <img alt="udeA logo" height="140px" src="https://github.com/freddyduitama/images/blob/master/logo.png?raw=true" style="width:107px;height:152px; margin-right: 40px;">
    <div style="text-align: center;">
        <h1><b><font color='0B5345'>Sistema Predictor de grupos relacionados por el diagnóstico</font></b></h1>
        <h1><b><font color='0B5345'>Modelo Baseline</font></b></h1>
        <h2><font color='0B5345'>Luisa Enciso - María Ostos - Santiago Rivera</font></h2>
    </div>
</div>


In [8]:
import pandas as pd
import numpy as np
import unicodedata
import matplotlib.pyplot as plt

In [9]:
def mostrar_informe(df):
    print("Informe estadístico de la base de datos")
    print("="*50, "\n")

    # 1. Información general
    print(f"Número de registros: {df.shape[0]}")
    print(f"Número de columnas: {df.shape[1]}\n")

    # 2. Valores nulos por columna
    print("Valores nulos por columna:")
    print(df.isna().sum())
    print("\n")

    # 3. Valores únicos por columna
    print("Número de valores únicos por columna:")
    print(df.nunique())
    print("\n")

    # 4. Estadísticas descriptivas para columnas numéricas
    print("Estadísticas descriptivas (numéricas):")
    print(df.describe())
    print("\n")

    # 5. Frecuencias para variables categóricas
    cat_cols = df.select_dtypes(include=['object', 'category']).columns
    if len(cat_cols) > 0:
        print("Frecuencias para variables categóricas:")
        for col in cat_cols:
            print(f"\nColumna: {col}")
            print(df[col].value_counts(dropna=False))
    else:
        print("No se encontraron columnas categóricas.")

<hr>

# **Inspección de la base de datos**

In [10]:
# Ruta del archivo Excel
archivo_excel = 'DataBase.xlsx'
# Cargar el archivo Excel
df = pd.read_excel(archivo_excel, engine='openpyxl')
# Mostrar las primeras filas
df.head()

Num caso  Aseguradora -Código- Aseguradora -Descripción-  Edad Grupo Edad  \
0  413799.0                   204            Aseguradora_29    38      35-39   
1  412359.0                   313            Aseguradora_56    27      25-29   
2  412552.0                   313            Aseguradora_56    39      35-39   
3  414131.0                   227            Aseguradora_46    24      20-24   
4  410573.0                   227            Aseguradora_46    60      60-64   

  Sexo  Codigo de ciudad  Fecha de ingreso Tipo de ingreso  Fecha de egreso  \
0    F              5001          30052005        URGENCIA         19062005   
1    M              5088          11032005        URGENCIA         10042005   
2    M              5001          20032005        URGENCIA         10042005   
3    M              5001          20062005        URGENCIA         22062005   
4    M              5001          24112004        URGENCIA          7012005   

   ...  Proc28  Proc29 Proc30     Tipo servicio  Causa Basica de muerte  \
0  ...     NaN     NaN    NaN  URGENCIA ADULTOS                     NaN   
1  ...     NaN     NaN    NaN         NO APLICA                     NaN   
2  ...     NaN     NaN    NaN  URGENCIA ADULTOS                     NaN   
3  ...     NaN     NaN    NaN  URGENCIA ADULTOS                     NaN   
4  ...     NaN     NaN    NaN  URGENCIA ADULTOS                     NaN   

  Infecciones Infección Quirurgica GRD -Código  \
0          NO                   NO       14143   
1          NO                   NO      184153   
2          NO                   NO      184153   
3          NO                   NO      184153   
4          NO                   NO      184112   

                                    GRD -Descripción Tipo GRD  
0   MH ACCIDENTE VASCULAR CEREBRAL CON INFARTO w/MCC        M  
1  MH INFECCIÓN POR VIRUS DE INMUNODEFICIENCIA HU...        M  
2  MH INFECCIÓN POR VIRUS DE INMUNODEFICIENCIA HU...        M  
3  MH INFECCIÓN POR VIRUS DE INMUNODEFICIENCIA HU...        M  
4  MH INFECCIONES POST OPERATORIAS Y POSTRAUMÁTIC...        M  

[5 rows x 68 columns]

## **Informe Estadístico Inicial de la Base de Datos**

In [11]:
df_informe = pd.read_excel(archivo_excel, engine='openpyxl')
mostrar_informe(df_informe)

Informe estadístico de la base de datos

Número de registros: 27869
Número de columnas: 68

Valores nulos por columna:
Num caso                     1
Aseguradora -Código-         0
Aseguradora -Descripción-    0
Edad                         0
Grupo Edad                   0
                            ..
Infecciones                  0
Infección Quirurgica         0
GRD -Código                  0
GRD -Descripción             0
Tipo GRD                     0
Length: 68, dtype: int64


Número de valores únicos por columna:
Num caso                     7900
Aseguradora -Código-           62
Aseguradora -Descripción-      71
Edad                          108
Grupo Edad                     20
                             ... 
Infecciones                     4
Infección Quirurgica            4
GRD -Código                   608
GRD -Descripción              608
Tipo GRD                        4
Length: 68, dtype: int64


Estadísticas descriptivas (numéricas):
            Num caso  Aseguradora -

## 1. Descripción General

- **Número total de registros:** 27,869
- **Número total de columnas:** 68

Este informe presenta un análisis descriptivo preliminar con el fin de entender la estructura, calidad y características generales del conjunto de datos.



## 2. Calidad de los Datos

- **Valores nulos por columna:**
    
    La mayoría de las columnas no presentan valores nulos. Se identifica 1 registro con valor nulo en la columna `Num caso`.
      



## 3. Distribución de Datos Categóricos

Se presentan las frecuencias de las principales variables categóricas para identificar posibles valores atípicos o inconsistencias, además de observar la distribución general:

- **Sexo:**
    - Masculino (M): 14,353
    - Femenino (F): 13,512
    - Valores inconsistentes (m, f): 4 (corregir para homogeneizar)
- **Tipo de ingreso:**
    - Urgencia: 25,570
    - Programado: 2,294
    - Valores inconsistentes (variaciones en mayúsculas/minúsculas): corregir para uniformidad
- **Cuidados intensivos:**
    - NO: 26,681
    - SI: 1,182
    - Valores inconsistentes (variaciones en mayúsculas/minúsculas): corregir
- **Grupo Etario:**
    
    Distribución detallada por rangos de edad, destacando grupos mayores (75-79, 80-84 años) con mayor frecuencia.
    
- **Aseguradora:**
    
    71 categorías, con 5 aseguradoras predominantes que concentran la mayoría de los registros.
    
- **Diagnósticos (Dx principal y Dxr 1-9):**
    
    Amplia variedad con valores NaN significativos en las últimas columnas de diagnósticos secundarios.
    


## 4. Estadísticas Descriptivas para Variables Numéricas

- **Edad:**
    - Media: 57.7 años
    - Rango: 0 a 109 años
    - Desviación estándar: 24.17
- **Días de estancia hospitalaria:**
    - Media: 8.45 días
    - Rango: 0 a 174 días
    - Mediana: 6 días
- **Días en Unidad de Cuidados Intensivos:**
    - Media: 0.38 días
    - Alta dispersión con máximo de 92 días, pero la mayoría sin estancia en UCI.
    
- **Variables de Procedimientos (Proc22 a Proc30):**
    
    Presentan muchos valores faltantes (solo disponibles en pocas observaciones), con medias alrededor de 83-89 y alta variabilidad en algunos casos. Se debe evaluar relevancia para análisis futuros.
    



## 5. Observaciones Generales
- Unificar formatos en variables categóricas (ej. mayúsculas/minúsculas).
- Tratar valores faltantes, en particular en columnas diagnósticas y procedimientos.
- Validar posibles registros duplicados.
- Eliminar categorías muy específicas o poco frecuentes en variables categóricas para evitar ruido.



## 6. Conclusiones preliminares

El dataset es amplio y contiene información clínica valiosa. Se evidencia buena calidad general en términos de datos faltantes, salvo excepciones puntuales. La distribución etaria y los diagnósticos principales presentan un perfil coherente para un estudio de análisis de gestión hospitalaria. Sin embargo, es imprescindible realizar un proceso de limpieza y transformación para garantizar la calidad y precisión en análisis posteriores.